src/Text/Pandoc/Readers/JATS.hs

```diff
@@ -161,23 +161,42 @@ parseBlock (Text (CData _ s _)) = if T.all isSpace s
                                      then return mempty
                                      else return $ plain $ trimInlines $ text s
 parseBlock (CRef x) = return $ plain $ str $ T.toUpper x
-parseBlock (Elem e) =
+parseBlock (Elem e) = do
+  sectionLevel <- gets jatsSectionLevel 
+  let parseBlockWithHeader = parseWithHeader (sectionLevel+1) (getBlocks e)
+
   case qName (elName e) of
         "p" -> parseMixed para (elContent e)
         "code" -> codeBlockWithLang
         "preformat" -> codeBlockWithLang
-        "disp-quote" -> parseBlockquote
-        "list" -> case attrValue "list-type" e of
-                    "bullet" -> bulletList <$> listitems
-                    listType -> do
-                      let start = fromMaybe 1 $
-                                  (filterElement (named "list-item") e
-                                               >>= filterElement (named "label"))
-                                   >>= safeRead . textContent
-                      orderedListWith (start, parseListStyleType listType, DefaultDelim)
-                        <$> listitems
-        "def-list" -> definitionList <$> deflistitems
-        "sec" -> gets jatsSectionLevel >>= sect . (+1)
+        "disp-quote" -> parseWithHeader (sectionLevel+1) parseBlockquote
```

Comment:
I think this logic is correct and well thought out, but jus a suggestion, since this had me wondering for a while: Would you call the new function something like  `wrapWithHeader` instead of `parseWithHeader`? 

This is because you now cleverly outsource the actual block parsing out of the function, and it now expects the (would-be) parsed block as a parameter. I know we discussed the fact that the execution of `getBlocks` actually happens inside the `parseWithHeader` function, but calling it `parseWithHeader` is a bit confusing, specially when you look at these three cases, `disp-quote`, `list`, `def-list`,  where you seem to have to parse the blocks twice: one to prepare the parameter, and a second time inside the actual function. 

Again, this is really just a suggestion, and this won't have an impact on output, but for the sake of clarity might help others understand what is happening. Up to you :)

Reply:
That makes sense! I'll make that change 👍